src/redux/slices/auth.ts

```diff
@@ -1,12 +1,97 @@
-// below rule can be safely turned off thanks to immer used by redux toolkit
-/* eslint-disable no-param-reassign */
-
-// Reducers can directly reassign state, or return a new state, BUT NOT BOTH
-// Detailed docs on state reassignment in reducers here https://redux-toolkit.js.org/api/createreducer/
-// There are pitfalls to immer however, which you can read here https://immerjs.github.io/immer/docs/pitfalls
-
-// TODO: Setup axios, encrypt, appConfig in utils
-// TODO: Create auth slices
-// import axios from 'utils/axios'
-// import { encrypt } from 'utils/encryption';
-// import { appConfig } from 'utils/config';
+import { createSlice, Dispatch } from '@reduxjs/toolkit'
+import { UseMutateAsyncFunction } from '@tanstack/react-query'
+import {
+  CreateAuthLoginProps,
+  CreateAuthLoginResponse
+} from '~/features/auth/types/login'
+import { appConfig } from '~/utils/config'
+import { encrypt } from '~/utils/encryption'
+import { User } from '~/features/user/types/user'
+
+export interface InitialState {
+  user: User
+  loggingIn: boolean
+  loginError: string
+  isAuthenticated: boolean
+}
+
+const initialState: InitialState = {
+  user: {
+    id: 0,
+    email: '',
+    firstName: '',
+    gender: '',
+    isAdmin: false,
+    isCompany: false,
+    isVerified: false,
+    lastName: '',
+    username: '',
+    phone: 0,
+    subscriptionId: 0
+  },
+  loggingIn: false,
+  loginError: '',
+  isAuthenticated: false
+}
+
+const { actions, reducer } = createSlice({
+  name: 'auth',
+  initialState,
+  reducers: {
+    login: (state) => {
+      state.loggingIn = true
+      state.loginError = ''
+    },
+    validAuth: (state, action) => {
+      state.user = action.payload
+      state.loggingIn = false
+      state.loginError = ''
+      state.isAuthenticated = true
+    },
+    invalidAuth: (state, action) => {
+      state.user = initialState.user
+      state.loggingIn = false
+      state.loginError = action.payload
+      state.isAuthenticated = false
+    }
```

Comment:
```suggestion
    validateAuth: (state, action) => {
      state.user = action.payload
      state.loggingIn = false
      state.loginError = ''
      state.isAuthenticated = true
    },
    invalidateAuth: (state, action) => {
      state.user = initialState.user
      state.loggingIn = false
      state.loginError = action.payload
      state.isAuthenticated = false
    }
```

ganti namingnya gini aja ya